File: backend/src/main/java/com/senior/project/backend/domain/Interest.java
package com.senior.project.backend.domain;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.*;

@Entity
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Generated
public class Interest {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="student_details_id")
    @JsonIgnore
    private StudentDetails studentDetails;
}

File: frontend/crd/src/domain/Interest.ts
export interface InterestJSON{
    id: string;
    name: string;
    studentDetailsID: string;
}

export class Interest{

    constructor(json: InterestJSON){
        this.id = json.id;
        this.name = json.name;
        this.studentDetailsID = json.studentDetailsID;
    }

    id: string;
    name: string;
    studentDetailsID: string;

    static makeEmpty(){
        return new Interest({
            id: '',
            name: '',
            studentDetailsID: ''
        });
    }
}